src/ripple/core/SQLInterface.h

```diff
@@ -0,0 +1,247 @@
+        DatabaseType dbtype,
+        TableType type,
+        LedgerIndex ledgerSeq) = 0;
+
+    virtual void
+    deleteBeforeLedgerSeq(
+        DatabaseType dbtype,
+        TableType type,
+        LedgerIndex ledgerSeq) = 0;
+
+    virtual int
+    getRows(DatabaseType dbtype, TableType type) = 0;
+
+    virtual struct RowsMinMax
+    getRowsMinMax(DatabaseType dbtype, TableType type) = 0;
+
+    virtual bool
+    saveValidatedLedger(
+        Application& app,
+        std::shared_ptr<Ledger const> const& ledger,
+        bool current) = 0;
+
+    virtual std::optional<SQLLedgerInfo>
+    loadLedgerInfoByIndex(
+        LedgerIndex ledgerSeq,
+        beast::Journal j) = 0;
+
+    virtual std::optional<SQLLedgerInfo>
+    loadLedgerInfoByIndexSorted(
+        bool ascendSort,
+        beast::Journal j) = 0;
+
+    virtual std::optional<SQLLedgerInfo>
+    loadLedgerInfoByIndexLimitedSorted(
+        LedgerIndex ledgerFirstIndex,
+        bool ascendSort,
+        beast::Journal j) = 0;
+
+    virtual std::optional<SQLLedgerInfo>
+    loadLedgerInfoByHash(
+        uint256 const& ledgerHash,
+        beast::Journal j) = 0;
+
+    virtual uint256
+    getHashByIndex(LedgerIndex ledgerIndex) = 0;
+
+    virtual std::optional<Hashes>
+    getHashesByIndex(
+        LedgerIndex ledgerIndex,
+        beast::Journal j) = 0;
+
+    virtual std::map<LedgerIndex, Hashes>
+    getHashesByIndex(
+        LedgerIndex minSeq,
+        LedgerIndex maxSeq,
+        beast::Journal j) = 0;
+
+    virtual Json::Value
+    loadTxHistory(Application& app, LedgerIndex startIndex) = 0;
+
+    virtual NetworkOPs::AccountTxs
+    getAccountTxs(
+        Application& app,
+        LedgerMaster& ledgerMaster,
+        AccountID const& account,
+        std::int32_t minLedger,
+        std::int32_t maxLedger,
+        bool descending,
+        std::uint32_t offset,
+        int limit,
+        bool bUnlimited,
+        beast::Journal j) = 0;
+
+    virtual std::vector<NetworkOPs::txnMetaLedgerType>
+    getAccountTxsB(
+        Application& app,
+        AccountID const& account,
+        std::int32_t minLedger,
+        std::int32_t maxLedger,
+        bool descending,
+        std::uint32_t offset,
+        int limit,
+        bool bUnlimited,
+        beast::Journal j) = 0;
+
+    virtual std::optional<NetworkOPs::AccountTxMarker>
+    accountTxPage(
+        AccountIDCache const& idCache,
+        std::function<void(std::uint32_t)> const& onUnsavedLedger,
+        std::function<
+            void(std::uint32_t, std::string const&, Blob&&, Blob&&)> const&
+            onTransaction,
+        AccountID const& account,
+        std::int32_t minLedger,
+        std::int32_t maxLedger,
+        bool forward,
+        std::optional<NetworkOPs::AccountTxMarker> const& marker,
+        int limit,
+        bool bAdmin,
+        std::uint32_t page_length) = 0;
+
+    virtual std::pair<boost::variant<Transaction::pointer, bool>, error_code_i>
+    loadTransaction(
```

Comment:
I'm not going to insist on this, and maybe this belongs in a different PR, but I think this function should return `STTx` in place of `Transaction`. `STTx` is more efficient than `Transaction` in the case where we only want the binary form of the transaction. However, changing the return type requires changing other code, so if that seems too complicated for this PR, I'm fine leaving the return type as-is.

Reply:
I think will be better to remain as is.